Language design restructure (#47)
chhoumann/p4: Report/sections/2_design/keywords.tex

\subsection{Keywords}\label{sec:keywords}
\Tabref{table:KeyWordsTable} introduces the reserved keywords in \dazel.  A convention has been introduced that defines all block-indicators to be denoted by an upper-case first character.

\begin{table}[h!]
    \centering
    \begin{tabular}{l|l}
    \textbf{Keyword}             & \textbf{Description}                                                                                                                                                                                                                                            \\ 
    \hline

>     Player                       & \begin{tabular}[c]{@{}l@{}}Is globally accessible and always \\points to the player. This allows \\the programmer to~create \\interactions involving~the player \\as well providing access to \\their properties such as~health \\and move speed.\end{tabular}  \\ 

Reviewer: ```suggestion
    Player                       & \begin{tabular}[c]{@{}l@{}}A globally accessible reference\\ to the player. This allows the\\ programmer to create interactions \\involving the player's properties such \\as health or position.\end{tabular}  \\ 
```

The description confuses me a little, so I've made a suggestion that explains how I view the player keyword.

Reviewer 2: You don't really create interactions with the player's "health, position, etc.", you have access to them as properties though, which the original text stated.

Reviewer: You create interactions _involving_ the player's properties. Why else would a property be accessible than to interact with it? Nevertheless, wording here is tricky. I'm not sure that I have the answer. We're  creating one element that is OOP-like (object) but our language does not support OOP.

Reviewer 2: LGTM!

    \hline
    Data                         & \begin{tabular}[c]{@{}l@{}}Denotes a data block which can define\\properties such as health on entities.\end{tabular}                                                                                                                                           \\ 
    \hline
    Pattern                      & \begin{tabular}[c]{@{}l@{}}Denotes a pattern block which can be used\\to define specific movement patterns\\that may be reused across\\multiple entities.\end{tabular}                                                                                          \\ 
    \hline
    Entities                     & \begin{tabular}[c]{@{}l@{}}Denotes an entities block which can be\\used to define placement of entities on\\a given screen.\end{tabular}                                                                                                                       
    \end{tabular}

>     \caption{Table containing reserved keywords in \dazel}

Reviewer: ```suggestion
    \caption{Table explaining reserved keywords in \dazel}
```
